src/node_file.cc

```diff
@@ -130,6 +130,28 @@ FileHandleReadWrap::~FileHandleReadWrap() {}
 
 FSReqBase::~FSReqBase() {}
 
+void FSReqBase::Cancel(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  FSReqCallback* self;
+  ASSIGN_OR_RETURN_UNWRAP(&self, args.Holder());
+  self->ReqWrap::Cancel();
+
+  Local<Value> argv[] = {
+      FIXED_ONE_BYTE_STRING(self->env()->isolate(), "Operation was cancelled."),
+      FIXED_ONE_BYTE_STRING(self->env()->isolate(), "AbortError"),
+  };
+  Local<Value> exception;
+  Local<Function> domexception_ctor;
+  if (!GetDOMException(self->env()->context()).ToLocal(&domexception_ctor) ||
+      !domexception_ctor
+           ->NewInstance(self->env()->context(), arraysize(argv), argv)
+           .ToLocal(&exception)) {
+    return;
+  }
+
+  self->Reject(exception);
+  args.GetReturnValue().SetUndefined();
```

Comment:
I think you can drop this line, as the `undefined` is the implicit default return value:

```suggestion
```